### PR TITLE
Enable passing a sequence for the `index` argument to `.list.get()`

### PR DESCRIPTION
**Pull request by shwina**
Closes https://github.com/rapidsai/cudf/issues/10552.

Depends on #10547

### CODE REVIEW

**Comment by bdice** [comment]
@shwina I'm going to wait until we merge #10547 to review this PR.

File: python/cudf/cudf/core/column/lists.py
Comment (bdice) [inline python/cudf/cudf/core/column/lists.py:402]: What happens if `len(index) != len(self)`? Is there a clear error?

At first, I misunderstood what this method would do. I thought passing a sequence of ints would extract multiple values, so that `s.list.get([0, 1])` would return a lists column with the first two elements of each contained list. This is really more like a segmented gather. The docs could be improved by specifying that the input indices correspond to one index per list row.

File: python/cudf/cudf/core/column/lists.py
Comment (bdice) [inline python/cudf/cudf/core/column/lists.py:407]: Can we use the unary negative operator `-index` instead of `np.negative`? My expectation was that a Column would have basic operations like that, just like Series. It seems like the comparator below `index >= lengths` works fine, so it surprises me that the unary negative isn't supported.

File: python/cudf/cudf/tests/test_list.py
Comment (bdice) [inline python/cudf/cudf/tests/test_list.py:321]: Can we also have a test where `index` is a Series object? That seems like a desirable use case since the `index` must be the same length as the input Series.

**Review by bdice — CHANGES_REQUESTED** [review]
I have a few comments below. Please rebase this now that the dependent PR has been merged.

File: python/dask_cudf/dask_cudf/tests/test_accessor.py
Comment (bdice) [inline python/dask_cudf/dask_cudf/tests/test_accessor.py:390]:
```suggestion
def test_get(data, index):
```

File: python/cudf/cudf/core/column/lists.py
Comment (vyasr) [reply python/cudf/cudf/core/column/lists.py]: Columns don't really have well-defined API boundaries. Historically we have invoked many of their internal methods from the Frame layer. Not all operators are currently supported, and some of my recent binops PRs have been working to improve that. Even `np.negative` probably only started working after #10509, which was just merged 2 weeks ago; prior to that I'm guessing that you would have had to do `-1 * index`. I think the action item here is to mark this as a TODO.

Comment (shwina) [reply python/cudf/cudf/core/column/lists.py]: I raised https://github.com/rapidsai/cudf/issues/10611.

File: python/cudf/cudf/core/column/lists.py
Comment (shwina) [reply python/cudf/cudf/core/column/lists.py:405]: I tried to make the docs more clear about this - could you take a look or suggest improvements?

File: python/cudf/cudf/tests/test_list.py
Comment (shwina) [reply python/cudf/cudf/tests/test_list.py:324]: Done

File: python/cudf/cudf/tests/test_list.py
Comment (bdice) [inline python/cudf/cudf/tests/test_list.py:328]: Duplicate test.
```suggestion
    assert_eq(cudf.Series([1, 4, 8]), sr.list.get(cudf.Series([0, 1, 2])))
```

**Review by bdice — APPROVED** [review]
I like the changes you made. I noted one duplicate test, otherwise LGTM! :)

File: python/cudf/cudf/_lib/lists.pyx
Comment (vyasr) [inline python/cudf/cudf/_lib/lists.pyx:129]: Why did this change? Not a problem but it seems extraneous.

File: python/cudf/cudf/core/column/lists.py
Comment (vyasr) [reply python/cudf/cudf/core/column/lists.py:405]: @bdice just in case you had any follow-ups here.

**Review by vyasr — APPROVED** [review]
LGTM. There are still a few unresolved conversations, nothing blocking but probably worth double-checking that everything there has been addressed now.

File: python/cudf/cudf/_lib/lists.pyx
Comment (shwina) [reply python/cudf/cudf/_lib/lists.pyx]: Fixed

**Review by charlesbluca — APPROVED** [review]

File: python/cudf/cudf/core/column/lists.py
Comment (bdice) [reply python/cudf/cudf/core/column/lists.py:406]: I'm happy with the outcome here. Sorry, I forgot to reply on this thread when I approved. :)

**Pull request merged**
